components/Post/PageProgress.tsx

```diff
@@ -5,17 +5,21 @@ const PageProgress = () => {
 
   useEffect(() => {
     const calculateScrollProgress = () => {
-      let percentage = (
-        ((document.body.scrollTop || document.documentElement.scrollTop) /
-          (document.documentElement.scrollHeight -
-            document.documentElement.clientHeight)) *
-        100
-      ).toPrecision(3);
+      const scrollHeight =
+        document.documentElement.scrollHeight - window.innerHeight;
+      const scrollTop =
+        (document.documentElement.scrollTop || document.body.scrollTop) /
+        scrollHeight;
+      const percentage = Math.round(scrollTop * 100);
 
-      progressBarRef.current.style.setProperty("width", percentage + "%");
+      progressBarRef.current.style.setProperty(
+        "width",
+        percentage > 100 ? "100%" : percentage + "%"
```

Comment:
need this to fix behavior when the client scroll to down the page progress bar it  is equal more then 100% and returns to 100%  in one second